src/files/compose/validateDappnodeCompose.ts

```diff
@@ -0,0 +1,237 @@
+import { Manifest } from "../manifest/types";
+import semver from "semver";
+import { ComposeService, Compose } from "./types";
+import { getIsCore } from "../../utils/getIsCore";
+import { params } from "./params";
+
+/**
+ * Validates against custom dappnode docker compose specs.
+ * This function must be executed after the official docker schema
+ * @param param0
+ */
+export function validateDappnodeCompose({
+  composeUnsafe,
+  manifest
+}: {
+  composeUnsafe: Compose;
+  manifest: Manifest;
+}): void {
+  const allowedNetworks = params.DOCKER_WHITELIST_NETWORKS.join(",");
```

Comment:
It's completely unnecessary to pass this as a parameter. `params` is available in all the module scope. The different names make this extra confusing. If you really think it's important to not compute `params.DOCKER_WHITELIST_NETWORKS.join(",")` again, at the top of the file just do

```ts
import { params } from "./params";

const DOCKER_WHITELIST_NETWORKS_CSV = params.DOCKER_WHITELIST_NETWORKS.join(",");
```